### DISCUSSION TITLE
get the pixel value

### DISCUSSION THREAD

**Discussion by ruirigel** [ImageMagick/ImageMagick#4009] [Help] [answered]
Hello,
sorry if this is not the exact category for my question
if you can help me I would appreciate it.
I have seen some solutions that come close to what I want to do, but it's not exactly what I need. Via command line is there any way to get the all pixel value within a 0-255 range of an mono image so I can do some calculations later?
The image comes from a mono camera sensor which monitors the night sky.

Thank you

**Comment by snibgo** [comment]
What version of IM, on what platform?

> ...  but it's not exactly what I need.

What exactly do you need? Do you want binary values, one byte per pixel? Or do you want text decimal numbers such as "123" separated by white space?

What is the input format? It would help if you could show (or link to) a sample input.

**Reply by ruirigel** [reply]
Hello,
can be text decimal numbers and separated by white space.
The image that comes from the sensor comes in jpg format

**Comment by snibgo** [comment, marked as answer]
I suggest the "pgm:" format. For example:

```
magick in.jpg -colorspace Gray -depth 8 -compress None pgm:
```

The output will be something like:
```
P2
2 10
255
255 255
227 227
198 198
170 170
142 142
113 113
85 85
57 57
28 28
0 0
```
"P2" is the name of the format.

The next line gives the dimensions, width=2 and height=10.

The next line "255" is the maximum value that can be represented.

We then have one line per row of the image. Here, each row has two pixels.

**Reply by ruirigel** [reply]
I forgot to mention the platform. I am testing on ubuntu 64bit but after the tests I want to use on raspbian.
so, using the package for debian the command would be:

`convert in.jpg -colorspace Gray -depth 8 -compress None pgm:- `
